src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +162,171 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
                         idsToDelete.Add(queueItem.DocumentId);
                 }
             }
+        }
+        catch (ProduceException<string, byte[]> e)
+        {
+            if (e.Error.Code == ErrorCode.TopicAuthorizationFailed)
+                ThrowTopicAclAuthorizationException(e, currentTopic.Name);
 
-            _producer.CommitTransaction();
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Produce operation failed. ");
+            if (TryToGetAclErrorsDetails(currentTopic.Name, out var detailsMessage))
+                errorMessageBuilder.Append(detailsMessage);
+            
+            AbortTransactionAfterException(errorMessageBuilder);
```

Comment:
Why the tx abort is required after `ProduceException`? Is it always the case or we should do it only for particular errors? Note that you don't call it `if (e.Error.Code == ErrorCode.TopicAuthorizationFailed)`

Reply:
1. We should always abort transaction that has failed, which I've learned from my mistake. The erroneous transaction didn't abort by itself, and entering `PublishMessage` caused an "Invalid transaction state" error after calling `BeginTransaction`. From then on, I'm aborting transactions after errors to maintain the smoothness of the ETL. 
2. I do call abort inside the `ThrowTopicAclAuthorizationException` - I see that I should've mentioned it in the method name

Reply:
after remaking aborting transactions, I'm calling explicitly TryAbortTransaction here, also I've gotten rid of the `ThrowTopicAclAuthorizationException`

Reply:
https://github.com/ravendb/ravendb/pull/17577/commits/4480ef6e0448e40e1e6469b929f2397b82a35dca